### PR TITLE
GettingStarted: Improve styling and add tracking

### LINKED ISSUE #78451: Onboarding panel clips poorly on narrow screens
The onboarding panel on the default OSS homepage dashboard gets clipped poorly on narrow screens. It's especially noticable with the new docked menu.

image

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**
In some screen sizes parts of the cards were cut off. This is what is being improved by this PR. I also adjusted one of the button positions as it was far away from the related content and too narrow with some unrelated text. See original position below.
Screenshot 2024-01-12 at 09 25 20

Additionally I added tracking to the cards and the button that removes the panel.

**Why do we need this feature?**
Users should be able to see the full content and not be forced to search for certain elements.
We need to collect data since we are planning to replace this home page by an improved one.

**Who is this feature for?**
Everyone starting freshly in grafana.

**Which issue(s) does this PR fix?**:
Fixes #78451

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/panel/gettingstarted/GettingStarted.tsx [hunk 1]
```diff
@@ -95,9 +97,9 @@ export class GettingStarted extends PureComponent<PanelProps, State> {
             </Button>
             {currentStep === steps.length - 1 && (
               <Button
-                className={cx(styles.backForwardButtons, styles.previous)}
+                className={styles.backForwardButtons}
                 onClick={this.onPreviousClick}
-                aria-label="To advanced tutorials"
+                aria-label="To basic tutorials"
```
Comment (JoaoSilvaGrafana) [inline public/app/plugins/panel/gettingstarted/GettingStarted.tsx:102]: Good catch! 👏

File: public/app/plugins/panel/gettingstarted/GettingStarted.tsx [hunk 2]
```diff
@@ -107,9 +109,9 @@ export class GettingStarted extends PureComponent<PanelProps, State> {
             </div>
             {currentStep < steps.length - 1 && (
               <Button
-                className={cx(styles.backForwardButtons, styles.forward)}
+                className={styles.backForwardButtons}
                 onClick={this.onForwardClick}
```
Comment (JoaoSilvaGrafana) [inline public/app/plugins/panel/gettingstarted/GettingStarted.tsx:113]: Is it worth putting interactions here to see if anyone even sees this button? It took me months to realize there was a button there

**Review by JoaoSilvaGrafana — COMMENTED** [review]
The icons being gone looks good to me!

The change to have both buttons on the right kind of threw me off when I tried to use it locally coz Im used to the button moving so my mouse was halfway across the screen before I was like "wait... it's not there?". Personally I preferred it before because then I could understand it was some sort of carousel, this way the button to go "back" kind of looks like it collapses things? Plus in terms of accessibility the order of the button is still before the rest of the panel when you are tabbing through so we need to change that.
But if UX is ok with this I'd just say to fix the accessibility problem 👍

**Review by joshhunt — COMMENTED** [review]
In general, great fixes! 👍 ✅ 

I'm not a fan of the change in location of the 'back' button. I think having the button in the direction of the content made it easier to understand how the pages relate to each other spatially.

But, I don't feel super strongly. I don't think anyone really uses this panel anyway (which is why we should measure!), and I'm sure they would figure it out anyway.

**Comment by L-M-K-B** [comment]
Thanks @JoaoSilvaGrafana and @joshhunt for your feedback. Like João suggested I'll ask our UX people for feedback on the button.
cc @amy-super and @staton-hyse11

**Comment by L-M-K-B** [comment]
Here are a screen records from before and after moving the button:

**Before**
https://github.com/grafana/grafana/assets/48948963/6338c840-a0a4-40c7-9c66-8518c420a3bf

**After**
https://github.com/grafana/grafana/assets/48948963/e363b97c-b2fc-4aae-b4ef-d2c3adc0f8a2

**Comment by L-M-K-B** [comment]
After thinking about your comment, @JoaoSilvaGrafana, and consulting with @amy-super I moved the button back to the left side. Let's keep the old styling for now and I'll add some tracking on these buttons for the slider. Moving the button now would probably manipulate the data.

File: public/app/plugins/panel/gettingstarted/components/Step.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ interface Props {
 
 export const Step = ({ step }: Props) => {
   const styles = useStyles2(getStyles);
+  console.log('step:', step);
```
Comment (JoaoSilvaGrafana) [inline public/app/plugins/panel/gettingstarted/components/Step.tsx:18]: console log

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Looks good just need to remove the console 👏

**Comment by staton-hyse11** [comment]
I understand the reason for the move but this is typical behavior for carousels. I agree with the move to revert the button to its original location. 

Here is a Google shopping screenshot as an example
image

**Review by joshhunt — APPROVED** [review]

**Pull request merged**